src/USER-MISC/fix_pafi.cpp

```diff
@@ -0,0 +1,798 @@
+    sprintf(buffer,"Compute %s for fix hp does not calculate a local array",computename);
+    error->all(FLERR,buffer);
+  }
+  if (PathCompute->size_peratom_cols < domain->dimension*3) {
+    sprintf(buffer,"Compute %s for fix hp has %d < %d fields per atom",computename,PathCompute->size_peratom_cols,domain->dimension*3);
+    error->all(FLERR,buffer);
+  }
+  if (comm->me==0) {
+    if (screen) fprintf(screen,"fix hp compute name,style: %s,%s\n",computename,PathCompute->style);
+    if (logfile) fprintf(logfile,"fix hp compute name,style: %s,%s\n",computename,PathCompute->style);
+  }
+
+
+  respa_level_support = 1;
+  ilevel_respa = nlevels_respa = 0;
+
+  temperature = force->numeric(FLERR,arg[4]);
+  t_period = force->numeric(FLERR,arg[5]);
+  seed = force->inumeric(FLERR,arg[6]);
+  // TODO UNITS
+  gamma = 1. / t_period / force->ftm2v;
+  sqrtD = sqrt(1.) * sqrt(24.0*force->boltz/t_period/update->dt/force->mvv2e*temperature) / force->ftm2v;
+
+  // optional args
+  iregion = -1;
+  idregion = NULL; // not used
+  int iarg = 7;
+  while (iarg < narg) {
+    if (strcmp(arg[iarg],"region") == 0) {
+      if (iarg+2 > narg) error->all(FLERR,"Illegal fix hp command");
+      iregion = domain->find_region(arg[iarg+1]);
+      if (iregion == -1)
+        error->all(FLERR,"Region ID for fix hp does not exist");
+      int n = strlen(arg[iarg+1]) + 1;
+      idregion = new char[n];
+      strcpy(idregion,arg[iarg+1]);
+      iarg += 2;
+    } else if (strcmp(arg[iarg],"overdamped") == 0) {
+      od_flag = force->inumeric(FLERR,arg[iarg+1]);
+      iarg += 2;
+    } else if (strcmp(arg[iarg],"com") == 0) {
+      com_flag = force->inumeric(FLERR,arg[iarg+1]);
+      iarg += 2;
+    } else error->all(FLERR,"Illegal fix hp command");
+  }
+  force_flag = 0;
+
+  for(int i = 0; i < 10; i++) {
+    c_v[i] = 0.;
+    c_v_all[i] = 0.;
+  }
+  for(int i=0; i<5; i++) {
+    proj[i] = 0.0;
+    proj_all[i] = 0.0;
+  }
+  for(int i=0; i<4; i++) {
+    results[i] = 0.0;
+    results_all[i] = 0.0;
+  }
+  maxatom = 1;
+  memory->create(h,maxatom,3,"FixPAFI:h");
+
+  // initialize Marsaglia RNG with processor-unique seed
+  random = new RanMars(lmp,seed + comm->me);
+
+  // nve
+  dynamic_group_allow = 1;
+  time_integrate = 1;
+}
+
+/* ---------------------------------------------------------------------- */
+
+FixPAFI::~FixPAFI()
+{
+  if (copymode) return;
+  delete random;
+  delete [] idregion;
+  memory->destroy(h);
+}
+
+/* ---------------------------------------------------------------------- */
+
+int FixPAFI::setmask()
+{
+  int mask = 0;
+  mask |= POST_FORCE;
+  mask |= POST_FORCE_RESPA;
+  mask |= MIN_POST_FORCE;
+  mask |= INITIAL_INTEGRATE;
+  // nve
+  mask |= FINAL_INTEGRATE;
+  mask |= INITIAL_INTEGRATE_RESPA;
+  mask |= FINAL_INTEGRATE_RESPA;
+  return mask;
+}
+
+/* ---------------------------------------------------------------------- */
+
+void FixPAFI::init()
+{
+  // set index and check validity of region
```

Comment:
you should check here, if the "PathCompute" compute is still valid. it may have been deleted since creating the fix.